### PR TITLE
docs: burning gas + gas flow diagrams

### PR DESCRIPTION
**Pull request by jakmeier**

### CODE REVIEW

File: docs/architecture/how/gas.md [hunk 1]
```diff
@@ -46,6 +54,63 @@ instead of once at the start. However, remember that the execution may happen on
...
+entire fee is burnt because there is no [priority
+fee](https://ethereum.org/en/developers/docs/gas/#priority-fee) that Ethereum
+pays out to validators.
+
+The following diagram shows how gas flows through the execution of a
+transaction. The transaction consists of a function call performing a cross
+contract call, hence two function calls in sequence. (Note: This diagram is
+heavily simplified, more accurate diagrams are further down.)
+
+![Very Simplified Gas Flow Diagram](https://user-images.githubusercontent.com/6342444/265536127-5934362f-18d1-4653-8dc6-e930af57ebc5.svg)
```
Comment (nikurt) [inline docs/architecture/how/gas.md:76]: Typo: aline.near instead of alice.near
image

File: docs/architecture/how/gas.md [hunk 2]
```diff
@@ -46,6 +54,63 @@ instead of once at the start. However, remember that the execution may happen on
...
+   `action_function_call` fee, are burnt 100%.
+2. The remainder of the burnt gas is multiplied by the runtime configuration
+   parameter
+   [`burnt_gas_reward`](https://github.com/near/nearcore/blob/master/core/primitives/res/runtime_configs/parameters.snap#L5C5-L5C5)
+   which currently is at 30%.
+
+During receipt execution, nearcore code tracks the `gas_burnt_for_function_call`
+separately from other gas burning to enable this contract reward calculations.
+
+In the (still simplified) flow diagram, the contract reward looks like this. For brevity, `gas_burnt_for_function_call` in the diagram is denotes as `wasm fee`.
```
Comment (nikurt) [inline docs/architecture/how/gas.md:108]:
```suggestion
In the (still simplified) flow diagram, the contract reward looks like this.
For brevity, `gas_burnt_for_function_call` in the diagram is denoted as `wasm fee`.
```

File: docs/architecture/how/gas.md [hunk 3]
```diff
@@ -46,6 +54,63 @@ instead of once at the start. However, remember that the execution may happen on
...
+
+![Slightly Simplified Gas Flow Diagram](https://user-images.githubusercontent.com/6342444/265536284-cc3cadf2-38cd-4b59-ab54-bcd9d4cad2e4.svg)
```
Comment (nikurt) [inline docs/architecture/how/gas.md:110]: Same typo: `s/aline/alice/`

**Review by nikurt — APPROVED** [review]

File: docs/architecture/how/gas.md (hunk 1 above)
Comment (jakmeier) [reply docs/architecture/how/gas.md]: Thanks! Somehow the :fire: also didn't display correctly. Should be fixed now.

**Pull request merged**